src/createServer.js

```diff
@@ -1,8 +1,73 @@
+/* eslint-disable no-console */
 'use strict';
 
+const http = require('http');
+const zlib = require('zlib');
+const fs = require('fs');
+const path = require('path');
+const formidable = require('formidable');
+const { pipeline } = require('stream');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer((req, res) => {
+    const urlNormalized = new URL(req.url, `http://${req.headers.host}`);
+    const form = new formidable.IncomingForm();
+    const compressors = {
+      gzip: zlib.createGzip(),
+      deflate: zlib.createDeflate(),
+      br: zlib.createBrotliCompress(),
+    };
+
+    if (urlNormalized.pathname === '/compress' && req.method === 'POST') {
+      form.parse(req, (error, { compressionType }, { file }) => {
+        if (error || !compressionType || !file) {
+          res.writeHead(400, { 'Content-type': 'text/plain' });
+
+          return res.end('Invalid form data');
```

Comment:
Great work 🚀 to improve readability and maintainability you should create variable to store all possible messages